stdlib/Random/src/XoshiroSimd.jl

```diff
@@ -0,0 +1,350 @@
+        @eval @inline _rotl23(x::NTuple{$N, VecElement{UInt64}}) = Core.Intrinsics.llvmcall($code,
+            NTuple{$N, VecElement{UInt64}},
+            Tuple{NTuple{$N, VecElement{UInt64}}},
+            x)
+
+        code = """
+        %res = add <$N x i64> %1, %0
+        ret <$N x i64> %res
+        """
+        @eval @inline _plus(x::NTuple{$N, VecElement{UInt64}}, y::NTuple{$N, VecElement{UInt64}}) = Core.Intrinsics.llvmcall($code,
+            NTuple{$N, VecElement{UInt64}},
+            Tuple{NTuple{$N, VecElement{UInt64}}, NTuple{$N, VecElement{UInt64}}},
+            x, y)
+
+        code = """
+        %res = xor <$N x i64> %1, %0
+        ret <$N x i64> %res
+        """
+        @eval @inline _xor(x::NTuple{$N, VecElement{UInt64}}, y::NTuple{$N, VecElement{UInt64}}) = Core.Intrinsics.llvmcall($code,
+            NTuple{$N, VecElement{UInt64}},
+            Tuple{NTuple{$N, VecElement{UInt64}}, NTuple{$N, VecElement{UInt64}}},
+            x, y)
+
+        code = """
+        %res = and <$N x i64> %1, %0
+        ret <$N x i64> %res
+        """
+        @eval @inline _and(x::NTuple{$N, VecElement{UInt64}}, y::NTuple{$N, VecElement{UInt64}}) = Core.Intrinsics.llvmcall($code,
+            NTuple{$N, VecElement{UInt64}},
+            Tuple{NTuple{$N, VecElement{UInt64}}, NTuple{$N, VecElement{UInt64}}},
+            x, y)
+
+        code = """
+        %res = or <$N x i64> %1, %0
+        ret <$N x i64> %res
+        """
+        @eval @inline _or(x::NTuple{$N, VecElement{UInt64}}, y::NTuple{$N, VecElement{UInt64}}) = Core.Intrinsics.llvmcall($code,
+            NTuple{$N, VecElement{UInt64}},
+            Tuple{NTuple{$N, VecElement{UInt64}}, NTuple{$N, VecElement{UInt64}}},
+            x, y)
+
+        code = """
+        %tmp = insertelement <1 x i64> undef, i64 %1, i32 0
+        %shift = shufflevector <1 x i64> %tmp, <1 x i64> %tmp, <$N x i32> zeroinitializer
+        %res = lshr <$N x i64> %0, %shift
+        ret <$N x i64> %res
+        """
+        @eval @inline _lshr(x::NTuple{$N, VecElement{UInt64}}, y::Int64) = Core.Intrinsics.llvmcall($code,
+            NTuple{$N, VecElement{UInt64}},
+            Tuple{NTuple{$N, VecElement{UInt64}}, Int64},
+            x, y)
+
+    end
+end
+
+
+
+
+
+
+#The mast values are used as res = _or(_and(res, mskBits), oneBits)
+
+mskBits(::Type{T}) where T = 0xffffffffffffffff
+mskBits(::Type{T}, ::Val{N}) where {T,N} =  ntuple(i-> VecElement(mskBits(T)), Val(N))
+
+oneBits(::Type{T}) where T =  0x0000000000000000
+oneBits(::Type{T}, ::Val{N}) where {T,N} =  ntuple(i-> VecElement(oneBits(T)), Val(N))
+
+mskBits(::Type{Float64})  = Base.significand_mask(Float64)
+oneBits(::Type{Float64}) = Base.exponent_one(Float64)
+
+mskBits(::Type{Float32}) = Base.significand_mask(Float32) + UInt64(Base.significand_mask(Float32))<<32
+oneBits(::Type{Float32}) = Base.exponent_one(Float32) + UInt64(Base.exponent_one(Float32))<<32
+
+
+function forkRand(rng::Union{TaskLocal, Xoshiro}, ::Val{N}) where N
```

Comment:
> How are you initializing the parallel instances? This is something that should be done with some care (e.g., using long jumps).
- @vigna 

Good point. It seems to be using `forkRand`.
FWIW, [VectorizedRNG.jl](https://github.com/chriselrod/VectorizedRNG.jl/blob/865757a23905686abff1202e83d5fafccb3ae295/src/xoshiro.jl#L37) does initialize them with jumps.